src/evict/evict_file.c

```diff
@@ -62,9 +62,17 @@ __wt_evict_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
          * sets of files open, updates in a no-longer-referenced file may not yet be visible, and
          * the write will fail with EBUSY. Our caller handles that error, retrying later.
          */
-        if (syncop == WT_SYNC_CLOSE && __wt_page_is_modified(page))
-            WT_ERR(__wt_reconcile(session, ref, NULL,
-              WT_REC_EVICT | WT_REC_HS | WT_REC_CLEAN_AFTER_REC | WT_REC_VISIBLE_ALL));
+        if (syncop == WT_SYNC_CLOSE && __wt_page_is_modified(page)) {
+            /*
+             * When setting the reconciliation flags, remember to not enable history store eviction
+             * for the history store file itself. Also metadata file doesn't have any associated
+             * history.
+             */
+            rec_flags = WT_REC_EVICT | WT_REC_CLEAN_AFTER_REC | WT_REC_VISIBLE_ALL;
+            if (!WT_IS_HS(btree) && !WT_IS_METADATA(dhandle))
+                rec_flags |= WT_REC_HS;
```

Comment:
What would be the symptom of this? Could it do something very odd?

Reply:
There is another path through which we set reconciliation flags when eviction that got me into a hung state. I discovered that after running the test script for a day. It is mentioned in the ticket. The path was:

`futex_reltimed_wait_cancelable,mutex=0x1f83068,,mutex=0x1f83068,,__wt_cond_wait_signal,__wt_cond_wait,__wt_readlock,__session_find_shared_dhandle,__session_get_dhandle,__wt_session_get_dhandle,__wt_session_get_btree_ckpt,__wt_curfile_open,__session_open_cursor_int,__wt_open_cursor,__hs_cursor_open_int,__wt_hs_cursor_open,__rec_hs_wrapup,__rec_write_wrapup,__reconcile,__wt_reconcile,__evict_review,__wt_evict,__wt_page_release_evict,__wt_page_in_func,__wt_page_swap_func,__wt_row_search,__cursor_row_search,__wt_btcur_insert,__curfile_insert,__wt_metadata_update,__wt_lsm_meta_write,__clsm_close_bulk,obj_bulk,fop,start_thread,clone`

The reason we started accessing HS when operating on metadata was that we were setting WT_REC_HS flag. That made me review all the uses for WT_REC_HS flag.

To answer your question, if we do not fix the flags in __wt_evict_file() we can possibly end up again in a hung state waiting to access dhandle.